### ISSUE TITLE
cannot reproduce table 3 for resnet 110

### ISSUE THREAD

**Issue by peymanrostami** [irwinherrmann/stochastic-gates#3]
Dear @irwinherrmann 

I find your work highly interesting and I am currently aiming to extend it further. I have edited the code slightly so that It now works on pytorch version 1.13.0 (though I suspect that the different version can slightly alter simulation results).

I would like to reproduce the result of table 3 for resnet 110.  
According to the paper, the "accelerated top-1 accuracy" should be 94.36%  (compared with 93.39 on the baseline) when using dependent gates with t=0.6 (65% flops). 
when reproducing the result, I run the code traincifar_resnetfilter.py with the same arguments as default, where [this line](https://github.com/irwinherrmann/stochastic-gates/blob/14828bf708831de0a0d5ca7b202403bb26b94ff9/traincifar_resnetfilter.py#L105) runs resnet110.

I don't have access to the pretrained weights of [cifar_cleanres111.pth.tar](https://github.com/irwinherrmann/stochastic-gates/blob/14828bf708831de0a0d5ca7b202403bb26b94ff9/traincifar_resnetfilter.py#L104), therefore I have to train from scratch. 

When training **from scratch**, the best accuracy after training is: 82.47.  In the paper however, it is reported as 94.36 which seems to be for the case **with the pretraining** of cifar_cleanres111.pth.tar. ?)

in another experiment, first, I trained the same network as above with dependent gates,  again from scratch while this time **setting t=1** (**i.e. all channels active during training)**, and I obtained: **Best accuracy: 84.7200.**
then, I used this trained network as a pretrained network and  trained it again for another 100 epochs keeping all the settings the same, but only returning t back to its original value of 0.6 (i.e. t=0.6), at the end of training, I obtained **Best accuracy: 84.6000**.
which means that by setting 60% of the activation off, the accuracy only decreases for 0.12 % **which is fascinating.**

Considering that I don't have access to the "cifar_cleanres111.pth.tar", is the result I am getting for the **training from scratch scenario** logical? 
Could you kindly provide me with the pretrained cifar_cleanres111.pth.tar file?

What other necessary steps (if any) should I take to reproduce the results of table 3 (in the paper) for resnet 110?

Your honor of helping me with this is highly demanded,

yours sincerely,
Peyman

**Comment by irwinherrmann** [comment]
Hi Peyman,

Sorry for the confusion, that model is a debugging model. In the model code [here](https://github.com/irwinherrmann/stochastic-gates/blob/14828bf708831de0a0d5ca7b202403bb26b94ff9/models/resnet_filter.py#L808), you can see that it's much smaller than a normal ResNet110. Please try: ResNet110_cifar.

Best,
Charles
